### ISSUE TITLE
Add detailed self-hosting instructions

### ISSUE THREAD

**Issue by mritunjayagarwal** [abi/screenshot-to-code#190]
Can anyone share me the steps to host this on my own domain? I am not very much familiar with docker and its deployments.

**Comment by abi** [comment]
You can use Docker to deploy this. But you can also just deploy it without Docker. The front-end is a simple react app, and the backend is a simple FastAPI Python server. I deployed it on Render for https://screenshottocode.com 

If someone wants to write up more detailed instructions, please do, and I'll add it to the README.

**Comment by mritunjayagarwal** [comment]
The server keeps crashing on Render

**Comment by abi** [comment]
Would Google the error

**Comment by mritunjayagarwal** [comment]
Its not like that. It boots up and then stops the script
I'll share my configurations

**Comment by mritunjayagarwal** [comment]
![Uploading Web capture_21-12-2023_20111_dashboard.render.com.jpeg…]()

**Comment by mritunjayagarwal** [comment]
![Web capture_21-12-2023_201047_dashboard render com](https://github.com/abi/screenshot-to-code/assets/39786312/375979c3-f4af-4957-b502-8532c8962469)

**Comment by mritunjayagarwal** [comment]
==> Deploying...
==> Using Node version 20.10.0 (default)
==> Docs on specifying a Node version: https://render.com/docs/node-version
==> Running 'poetry run uvicorn main:app --reload --port 7001'
INFO:     Will watch for changes in these directories: ['/opt/render/project/src']
INFO:     Uvicorn running on http://127.0.0.1:7001/ (Press CTRL+C to quit)
INFO:     Started reloader process [41] using StatReload
ERROR:    Error loading ASGI app. Could not import module "main".
Deploy cancelled
==> Deploying...
==> Using Node version 20.10.0 (default)
==> Docs on specifying a Node version: https://render.com/docs/node-version
==> Running 'poetry run uvicorn main:app --reload --port 7001'
INFO:     Will watch for changes in these directories: ['/opt/render/project/src/backend']
INFO:     Uvicorn running on http://127.0.0.1:7001/ (Press CTRL+C to quit)
INFO:     Started reloader process [43] using StatReload
INFO:     Started server process [47]
INFO:     Waiting for application startup.
INFO:     Application startup complete.

**Comment by mritunjayagarwal** [comment]
ERROR: Error loading ASGI app. Could not import module "main".

**Comment by abi** [comment]
This is my backend deployment on Render

<img width="1004" alt="Screenshot 2023-12-21 at 2 01 21 PM" src="https://github.com/abi/screenshot-to-code/assets/23818/af6df53a-92e9-4ef7-a0ce-3882c0922d3e">

env var

<img width="941" alt="Screenshot 2023-12-21 at 2 01 55 PM" src="https://github.com/abi/screenshot-to-code/assets/23818/fba4d9fb-ce8f-4155-885b-757cedcb03c8">

**Comment by abi** [comment]
Why are you trying to deploy it on Render instead of just running it locally? The docker set up should be easy

**Comment by mritunjayagarwal** [comment]
> This is my backend deployment on Render
> 
> <img alt="Screenshot 2023-12-21 at 2 01 21 PM" width="1004" src="https://private-user-images.githubusercontent.com/23818/292294043-af6df53a-92e9-4ef7-a0ce-3882c0922d3e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxODU4NjAsIm5iZiI6MTcwMzE4NTU2MCwicGF0aCI6Ii8yMzgxOC8yOTIyOTQwNDMtYWY2ZGY1M2EtOTJlOS00ZWY3LWEwY2UtMzg4MmMwOTIyZDNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIxVDE5MDYwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4ODRlMGMyODRmMWFmYjc5MGJhZDE1Y2YzYzNiOWI1NmNiZGVlZmRlYTI3NzY2YzM2Y2IzZDA2MTFmMDA3OTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.knn3d2FA8PAD9m0kaRJN7DUvjtpHjHWBFBEDs84oH6g">
> env var
> 
> <img alt="Screenshot 2023-12-21 at 2 01 55 PM" width="941" src="https://private-user-images.githubusercontent.com/23818/292294146-fba4d9fb-ce8f-4155-885b-757cedcb03c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxODU4NjAsIm5iZiI6MTcwMzE4NTU2MCwicGF0aCI6Ii8yMzgxOC8yOTIyOTQxNDYtZmJhNGQ5ZmItY2U4Zi00MTU1LTg4NWItNzU3Y2VkY2IwM2M4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIxVDE5MDYwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVlMzI2ZDU4Zjg1NjU5NjM0ZjI4NzQ5ZDMzMmE4YTZiZGY4OTdkOWE4YmUyNTZjZjIzM2E4MWE4OGQyMTNlNDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6pO-yhLaf-cpRqK4rrkCUbw9BAwZKYbmrT0rZodMX-U">

All other fields are default?

**Comment by abi** [comment]
Yes. Make sure root dir is `backend` and then you'd deploy `frontend` separately.

**Comment by mritunjayagarwal** [comment]
Sure. I am trying it

**Comment by mritunjayagarwal** [comment]
==> Cloning from https://github.com/mritunjayagarwal/screenshot_to_code-be
==> Checking out commit bbad73fafc093683660519f8a5d2a5ed1662d726 in branch main
==> Using Node version 20.10.0 (default)
==> Docs on specifying a Node version: https://render.com/docs/node-version
==> Using Python version 3.10.9 via environment variable PYTHON_VERSION
==> Docs on specifying a Python version: https://render.com/docs/python-version
==> Using Poetry version 1.7.1 (default)
==> Docs on specifying a Poetry version: https://render.com/docs/poetry-version
==> Running build command './build.sh'...
bash: ./build.sh: No such file or directory
==> Build failed 😞

**Comment by mritunjayagarwal** [comment]
Build succeeded.

==> Build uploaded in 9s
==> Build successful 🎉
==> Using Node version 20.10.0 (default)
==> Docs on specifying a Node version: https://render.com/docs/node-version
==> Running 'poetry run uvicorn main:app --reload --port 7001'
INFO:     Will watch for changes in these directories: ['/opt/render/project/src']
INFO:     Uvicorn running on http://127.0.0.1:7001/ (Press CTRL+C to quit)
INFO:     Started reloader process [41] using StatReload
ERROR:    Error loading ASGI app. Could not import module "main".


I am having this ASGI missing error.

**Comment by mritunjayagarwal** [comment]
Everything worked find.  @abi 
Only the timeout occurs  after 15mins and deployment is automatically cancelled.
Last log is application startup complete
Is there any other setting to stop this timeout

**Comment by mritunjayagarwal** [comment]
This issue is resolved

**Comment by abi** [comment]
Going to leave it open so someone can add in a doc with more detailed instructions.

Glad it's solved for you!
